PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/springcloud/dependency/action/AddAzureDependencyAction.java

```diff
@@ -75,48 +76,55 @@ public boolean onActionPerformed(@NotNull AnActionEvent event, @Nullable Operati
                     return;
                 }
+                final String springBootVer = getMavenLibraryVersion(mavenProject, SPRING_BOOT_GROUP_ID, "spring-boot-autoconfigure");
+                progressIndicator.setText("Get latest versions ...");
                 SpringCloudDependencyManager manager = new SpringCloudDependencyManager(evaluateEffectivePom);
-
                 Map<String, DependencyArtifact> versionMaps = manager.getDependencyVersions();
-                ProgressManager.checkCanceled();
-                DependencyArtifact springBootArtifact = versionMaps.get(SPRING_BOOT_GROUP_ID + ":spring-boot-autoconfigure");
-                if (springBootArtifact == null || StringUtils.isEmpty(springBootArtifact.getCurrentVersion())) {
-                    throw new AzureExecutionException(String.format("Module %s is not a spring-boot application.", module.getName()));
-                }
-                final String springBootVer = springBootArtifact.getCurrentVersion();
-                ProgressManager.checkCanceled();
-
-                progressIndicator.setText("Get latest versions ...");
                 List<DependencyArtifact> dep = new ArrayList<>();
                 dep.add(getDependencyArtifact(GROUP_ID, ARTIFACT_ID, versionMaps));
                 dep.add(getDependencyArtifact(SPRING_BOOT_GROUP_ID, "spring-boot-starter-actuator", versionMaps));
                 dep.add(getDependencyArtifact(SPRING_CLOUD_GROUP_ID, "spring-cloud-config-client", versionMaps));
-                dep.add(getDependencyArtifact(SPRING_CLOUD_GROUP_ID, "spring-cloud-starter-netflix-eureka-client", versionMaps));
+                dep.add(getDependencyArtifact(SPRING_CLOUD_GROUP_ID,
+                                              "spring-cloud-starter-netflix-eureka-client",
+                                              versionMaps));
                 dep.add(getDependencyArtifact(SPRING_CLOUD_GROUP_ID, "spring-cloud-starter-zipkin", versionMaps));
                 dep.add(getDependencyArtifact(SPRING_CLOUD_GROUP_ID, "spring-cloud-starter-sleuth", versionMaps));
                 ProgressManager.checkCanceled();
                 List<DependencyArtifact> versionChanges = manager.getCompatibleVersions(dep, springBootVer);
                 if (versionChanges.isEmpty()) {
                     PluginUtil.showInfoNotificationProject(project, "Your project is update-to-date.",
-                            "No updates are needed.");
+                                                           "No updates are needed.");
                     return;
                 }
-                ProgressManager.checkCanceled();
                 progressIndicator.setText("Applying versions ...");
                 File pomFile = new File(mavenProject.getFile().getCanonicalPath());
+                ProgressManager.checkCanceled();
+                Map<String, DependencyArtifact> managementVersions = manager.getDependencyManagementVersions();
```

Comment:
use stream API to filter the `DependencyArtifact` which `(StringUtils.equals(change.getCompatibleVersion(), managementVersion)`? So that no need to check whether version is empty below

Reply:
Done